File: tests/fixtures/vulnerable/Injection.java
package demo;

import java.sql.Connection;
import java.sql.ResultSet;
import java.sql.Statement;

public class Injection {

    // Vulnerable: SQL Injection (CG-001) — string concatenation
    public ResultSet getUser(Statement stmt, String name) throws Exception {
        return stmt.executeQuery("SELECT * FROM users WHERE name = '" + name + "'");
    }

    // Vulnerable: SQL Injection (CG-001) — String.format assembly
    public ResultSet findOrder(Statement stmt, String id) throws Exception {
        return stmt.executeQuery(String.format("SELECT * FROM orders WHERE id = %s", id));
    }

    // Vulnerable: SQL Injection (CG-001) — two-step String.format
    public void deleteUser(Connection conn, String name) throws Exception {
        String query = String.format("DELETE FROM users WHERE name = '%s'", name);
        conn.prepareStatement(query).execute();
    }

    // Vulnerable: Command Injection (CG-002) — Runtime.exec with concatenation
    public Process runTool(String dir) throws Exception {
        return Runtime.getRuntime().exec("ls -la " + dir);
    }

    // Vulnerable: Command Injection (CG-002) — ProcessBuilder with concatenation
    public ProcessBuilder ping(String host) {
        return new ProcessBuilder("sh", "-c", "ping -c 1 " + host);
    }
}

File: tests/fixtures/vulnerable/injection.go
package main

import (
	"database/sql"
	"fmt"
	"net/http"
	"os/exec"
)

// Vulnerable: SQL Injection (CG-001) — fmt.Sprintf query assembly
func getUser(db *sql.DB, name string) (*sql.Rows, error) {
	query := fmt.Sprintf("SELECT * FROM users WHERE name = '%s'", name)
	return db.Query(query)
}

// Vulnerable: SQL Injection (CG-001) — Sprintf inline
func findOrder(db *sql.DB, w http.ResponseWriter, r *http.Request) {
	id := r.URL.Query().Get("id")
	rows, _ := db.Query(fmt.Sprintf("SELECT * FROM orders WHERE id = %s", id))
	_ = rows
}

// Vulnerable: SQL Injection (CG-001) — string concatenation
func deleteUser(db *sql.DB, name string) error {
	_, err := db.Exec("DELETE FROM users WHERE name = '" + name + "'")
	return err
}

// Vulnerable: Command Injection (CG-002) — concatenated shell command
func runTool(dir string) ([]byte, error) {
	cmd := exec.Command("sh", "-c", "ls -la "+dir)
	return cmd.Output()
}

// Vulnerable: Command Injection (CG-002) — Sprintf-built command
func ping(host string) ([]byte, error) {
	cmd := exec.Command("sh", "-c", fmt.Sprintf("ping -c 1 %s", host))
	return cmd.Output()
}
